Extension Project_Final.Rmd

```diff
@@ -0,0 +1,768 @@
+
+corn$state.name <- na.locf(corn$state.name)
+
+corn <- corn %>%
+  filter(grepl("^\\s\\s\\s+|\\s+$", X1), grepl("MAIZ", product)) 
+
+#coffee
+coffee <-  read_excel("data/VIII Censo Agrícola 2007_Municipal.xls", 
+                  sheet= "Cuadro 10", skip = 12, col_names = FALSE) 
+
+coffee <- coffee %>%
+  filter(row_number() <= 10448) %>%
+  mutate(product = ifelse(!grepl("^\\s+|\\s+$", X1), X1, NA))
+
+coffee$product <- na.locf(coffee$product)
+coffee$product <- gsub("`|\\'", "", iconv(coffee$product, to="ASCII//TRANSLIT"))
+
+coffee <- coffee %>%
+  filter(grepl("^\\s+|\\s+$", X1)) %>% 
+  mutate(state.name = ifelse(!grepl("^\\s\\s\\s+|\\s+$", X1), X1, NA))
+
+coffee$state.name <- na.locf(coffee$state.name)
+coffee$state.name <- trim(coffee$state.name)
+
+coffee <- coffee %>%
+  filter(grepl("^\\s\\s\\s+|\\s+$", X1), grepl("CAFE", product)) 
+
+#cattle
+cattle <- read_excel("data/VIII Censo Agrícola 2007_Municipal.xls", 
+                   sheet= "Cuadro 31", skip = 11, col_names = FALSE) 
+
+cattle <- cattle %>%
+  filter(row_number() <= 2481) %>%
+  mutate(state.name = X1) %>%
+  select(-X1) 
+
+cattle$state.name <- na.locf(cattle$state.name)
+
+cattle <- cattle %>%
+  filter(!is.na(X2))
+
+colnames(cattle) <- c("X1", "X2", "X3", "X4", "X5", "X6", "state.name")
+
+#function to identify if the name is of a state or a municipality. If there are two leading white spaces it is a municipality, if there is only one white space it is a state and moved to a new column
+
+naming <- function(df) {
+
+df <- df %>%
+  rename(mun.name = X1)
+
+  ## standardizes state names
+  df$state.name <- gsub("`|\\'", "", iconv(df$state.name, to="ASCII//TRANSLIT"))
+  df$state.name <- gsub("COAHUILA DE ZARAGOZA", "COAHUILA", df$state.name)
+  df$state.name <- gsub("VERACRUZ LLAVE", "VERACRUZ", df$state.name)
+  df$state.name <- gsub("MICHOACAN DE OCAMPO", "MICHOACAN", df$state.name)
+
+  ## removes accents and tilde in municipal names
+  df$mun.name<- gsub("`|\\'", "", iconv(df$mun.name, to="ASCII//TRANSLIT"))
+  df$mun.name <- gsub("~","", df$mun.name)
+  df$mun.name <- gsub("¨","", df$mun.name)
+  df$mun.name <- gsub("~","", df$mun.name)
+  df$mun.name <- gsub("\"","", df$mun.name)
+
+
+  df$mun.name <- trim(df$mun.name)
+  df$state.name <- trim(df$state.name)
+  df <- df %>%
+    mutate(name = paste(mun.name, state.name, sep = ", "))
+
+}
+
+prod_units <- naming(prod_units)
+
+prod_units <- prod_units %>%
+  select(name, mun.name, state.name, total.units = X2, surface = X3)
+
+property_type <- naming(property_type)
+
+property_type <- property_type %>%
+  select(name, ejidal = X3, private = X5, comunal = X4)
+
+corn <- naming(corn)
+
+corn <- corn %>%
+  group_by(name) %>%
+  summarise(corn.tons = sum(X6), corn.ha = sum(X3))
+
+coffee <- naming(coffee)
+coffee$X2 <- as.numeric(as.character(coffee$X2))
+
+coffee <- coffee %>% 
+  group_by(name) %>%
+  summarise(total.coffee.units = sum(X2)) #lots of NAs here, from stars
+
```

Comment:
If you have a lot of NAs in your data and think that may change the results, double-checking  their impact through data imputations could be helpful. Maybe too much work for a replication project but it can be worthwhile for your own work.